src/building/count.c

```diff
@@ -59,6 +147,195 @@ int building_count_upgraded(building_type type)
     return upgraded;
 }
 
+int building_count_in_area(building_type type, int minx, int miny, int maxx, int maxy)
+{
+    int array_size = building_count();
+    int last_entry_position = 0;
+    int* found_buildings = (int*) malloc(array_size * sizeof(int));
+
+    int total = 0;
+    for (int x = minx; x <= maxx; x++) {
+        for (int y = miny; y <= maxy; y++) {
+            int grid_offset = map_grid_offset(x, y);
+            int building_id = map_building_at(grid_offset);
+            if (building_id) {
+                building *b = building_main(building_get(building_id));
+                if (type != BUILDING_ANY && b->type != type) {
+                    continue;
+                }
+                if (b->state != BUILDING_STATE_IN_USE && b->state != BUILDING_STATE_CREATED) {
+                    continue;
+                }
+
+                for (int i = 0; i < array_size; i ++) {
```

Comment:
Small performance save:

```suggestion
                for (int i = 0; i < total; i ++) {
```

Reply:
Fixed.